File: saxpy_array.jl
function saxpy(N)
    println("N: $N")
    
    const XVAL = Float32(10 * rand())
    const YVAL = Float32(10 * rand())
    const AVAL = Float32(10 * rand())
    
    x = zeros(Float32,N) + XVAL
    y = zeros(Float32,N) + YVAL

    time = @elapsed y += AVAL * x
    
    time = time * 1000
    println("Elapsed: $time ms")
    
    const TRUEVAL = YVAL + AVAL * XVAL
    err = sum(y - TRUEVAL)
    println("Error: $err")
    
end

saxpy(1 << 26)

File: saxpy_array.R
N <- 2 ^ 26
cat("N:", N, "\n")

# Random numbers
XVAL <- 10 * runif(1)
YVAL <- 10 * runif(1)
AVAL <- 10 * runif(1)

x <- array(XVAL, dim=c(N))
y <- array(YVAL, dim=c(N))

t0 <- Sys.time()
y <- y + x * AVAL
diff <- Sys.time() - t0
cat("Elapsed:", diff*1000, " ms\n")

answer <- YVAL + AVAL * XVAL
error <- sum(abs(y - answer))
cat("Error:", error, "\n")
